src/components/TreeGrid/TreeGrid.Props.ts

```diff
@@ -0,0 +1,28 @@
+import { GridColumn, SortDirection } from '../QuickGrid/QuickGrid.Props';
+
+export interface ITreeGridProps {
+    tree: Array<TreeNode>;
+    columns: Array<GridColumn>;
+    className?: string;
+    onRowDoubleClicked?: (row: any) => void;
+    onSelectedRowChanged?: (selectedRowIndex: number) => void;
+    sortColumn?: string;
+    sortDirection?: SortDirection;
+    columnSummaries?: any;
+}
+
+export interface ITreeGridState {
+    collapsedTreeNodes: Array<TreeNode>;
+    selectedRowIndex?: number;
+    columnsToDisplay: Array<GridColumn>;
+    sortColumn?: string;
+    sortDirection?: SortDirection;
+}
+
+
+export interface TreeNode { // extend this interface on a data structure to be used for row data
+    TreeId: string;
+    ParentId: string; // treeId of the parent node
+    IsExpanded: boolean;
+    leaves: Array<TreeNode>;
```

Comment:
leaves gives me a picture of only nodes on the lowest level. will the root node in a tree with 5 levels also have leaves.length > 0? If so then a better name would be children

Reply:
Indeed, changed